src/super_gradients/training/sg_trainer/sg_trainer.py

```diff
@@ -1318,17 +1339,39 @@ def forward(self, inputs, targets):
                     self.net = self.ema_model.ema
 
                 # RUN TEST ON VALIDATION SET EVERY self.run_validation_freq EPOCHS
+                valid_metrics_dict = {}
                 if (epoch + 1) % self.run_validation_freq == 0:
                     self.phase_callback_handler.on_validation_loader_start(context)
                     timer.start()
                     valid_metrics_dict = self._validate_epoch(context=context, silent_mode=silent_mode)
                     inf_time = timer.stop()
 
+                    self.valid_monitored_values = sg_trainer_utils.update_monitored_values_dict(
+                        monitored_values_dict=self.valid_monitored_values,
+                        new_values_dict=valid_metrics_dict,
+                    )  # TODO: Move this logic inside a MonitoredValues class
                     # Phase.VALIDATION_EPOCH_END
                     # RUN PHASE CALLBACKS
                     context.update_context(metrics_dict=valid_metrics_dict)
                     self.phase_callback_handler.on_validation_loader_end(context)
 
+                test_metrics_dict = {}
+                if (epoch + 1) % self.run_validation_freq == 0:
```

Comment:
If you decided that we also use run_validation_freq attribute to when we perform test you need to state this in the docs please.

Reply:
I added a new param similar to validation: https://github.com/Deci-AI/super-gradients/pull/1241/commits/ab9f715e1dac36c3b10ad5bcc47a12669bd71d7b

Reply:
(I changed the value to 1 here [7321c0c](https://github.com/Deci-AI/super-gradients/pull/1241/commits/7321c0ce0c6a7dc94ff2e67a86168514aed23f8f), that was a mistake)